cosmo_tester/framework/test_hosts.py

```diff
@@ -869,26 +870,50 @@ def _set_image_details(self):
             )
 
     def prepare_xfsdump_volume(self):
-        self._logger.info('Preparing XFS volume for XFS-dump')
+        self._logger.info('Preparing XFS dump volume')
         with self.ssh() as fabric_ssh:
             # Identify the still not partitioned volume
-            result = fabric_ssh.run(
-                "lsblk -o NAME,TYPE -ds | awk '$2 == \"disk\" {print $1}'")
-            self._logger.info('>>>> Disk name = `%s`', result.stdout())
+            volume_id = fabric_ssh.run(
+                "lsblk -o NAME,TYPE -ds | awk '$2 == \"disk\" {print $1}'"
+            ).stdout.strip()
+            if not volume_id:
+                self._logger.warning('No XFS dump volume found. '
+                                     'roceeding without.')
+                return
+            # Partition the volume and format for XFS
+            fabric_ssh.run("""echo "n
+p
+1
 
-            fabric_ssh.run(
-                "export XFSVOL=$(lsblk -o NAME,TYPE -ds "
-                "| awk '$2 == \"disk\" {print $1}')")
-
-#             result = fabric_ssh.run(
-#                 """echo "n
-# p
-# 1
-#
-#
-# w" | sudo fdisk /dev/$XFSVOL
-# sudo mkfs -t xfs /dev/${XFSVOL}p1 """)
 
+w" | sudo fdisk /dev/{}""".format(volume_id))
+            fabric_ssh.sudo("mkfs -t xfs /dev/{}p1 """.format(volume_id))
+            # Install XFS dump tool
+            fabric_ssh.sudo("yum install -y xfsdump")
+        self.xfsdump_volume_id = volume_id
+
+    def xfs_dump(self):
+        if self.xfs_restore_exists():
+            return
+        self._logger.info('Creating an XFS dump for host {}. Might take up to '
+                          '5 minutes...'.format(self.server_id))
+        with self.ssh() as fabric_ssh:
+            fabric_ssh.sudo('xfsdump -l0 -L test -M test -f /dev/{}p1 /'
+                            .format(self.xfsdump_volume_id))
+
+    def xfs_restore_exists(self):
```

Comment:
Might be best just setting this as a property on the VM- no sense SSHing in if the command to create the dump succeeded?

Reply:
Switching to reviewing as one block, not by commit- will re-raise comments as required.